src/index.js

```diff
@@ -1,34 +1,91 @@
-/* eslint-disable react/react-in-jsx-scope, react/jsx-filename-extension, no-unused-vars */
-
-/* @jsx createElement */
-
-function createElement(tagName, props, ...children) {
-  const element = document.createElement(tagName);
-
-  Object.entries(props || {}).forEach(([key, value]) => {
-    element[key.toLowerCase()] = value;
-  });
-
-  children.flat().forEach((child) => {
-    if (child instanceof Node) {
-      element.appendChild(child);
-      return;
-    }
-    element.appendChild(document.createTextNode(child));
-  });
-
-  return element;
-}
-
-function render() {
-  const element = (
-    <div>
-      <p>간단 계산기</p>
-    </div>
-  );
-
-  document.getElementById('app').textContent = '';
-  document.getElementById('app').appendChild(element);
-}
-
-render();
+/* eslint-disable react/react-in-jsx-scope, react/jsx-filename-extension, no-unused-vars */
+
+/* @jsx createElement */
+
+function createElement(tagName, props, ...children) {
+  const element = document.createElement(tagName);
+
+  Object.entries(props || {}).forEach(([key, value]) => {
+    element[key.toLowerCase()] = value;
+  });
+
+  children.flat().forEach((child) => {
+    if (child instanceof Node) {
+      element.appendChild(child);
+      return;
+    }
+    element.appendChild(document.createTextNode(child));
+  });
+
+  return element;
+}
+
+const makeDecimal = (accumulator, currentValue) => accumulator * 10 + currentValue;
+
+function calculate(operand1, operand2, operator) {
+  function add() {
+    console.log('+');
+    return operand1 + operand2;
+  }
+  function abstract() {
+    console.log('-');
+    return operand1 - operand2;
+  }
+  function multiply() {
+    console.log('*');
+    return operand1 * operand2;
+  }
+  function divide() {
+    console.log('/');
+    return operand1 / operand2;
+  }
+  function equal() {
+    console.log('=');
+    return 0;
+  }
+  return new Map([['+', add()], ['-', abstract], ['*', multiply], ['/', divide], ['=', equal]])[operator];
+}
+
+function render(input, num) {

```

Comment:
매개변수가 많아질 때 어떻게 관리할 수 있을까요? 확장성 측면에서 조금 고민해보시면 좋을 것 같아요~

Reply:
이 부분은 이번 커밋에서 Map은 사용하지 않았지만 이런 경우에 const로 저장해서 그 변수명을 넣고 작성할때 엔터로 가독성이 좋게 하는 방향이 나을것 같습니다. 맞는 접근인지 궁금합니다.

Reply:
아 render 함수의 매개변수를 말했었던 거에요~ Map까지 선택이 되어있었네요 😭

이런 경우에는 객체로 처리할 수 있을 것 같아요 ㅎㅎ
```suggestion
function render({
  presentNumber = '0',
  previousNumber = 'X',
  presentSign = 0,
  carrier = 0,
}) {
```
이렇게되면
1. named parameter처럼 파라미터에 이름을 붙이는 것처럼 표현력이 좋아집니다.
2. optional한 값을 다루기가 쉬워집니다.
3. 의미있는 파라미터들을 객체로 묶을 수 있습니다.

- https://github.com/hannut91/refactoring-practice/tree/master/6.8-introduce-parameter-object
- https://refactoring.com/catalog/introduceParameterObject.html